airbyte-ci/connectors/live-tests/src/live_tests/conftest.py

```diff
@@ -906,3 +927,23 @@ async def read_with_state_target_execution_result(
     request.config.stash[stash_keys.REPORT].add_target_execution_result(execution_result)
 
     return execution_result
+
+
+def respect_validation_test_mode(func: Callable) -> Any:
+    @functools.wraps(func)
+    async def wrapper(f, *args: Any, **kwargs: Any) -> Any:
+        try:
+            [config] = [a for a in list(args) + list(kwargs.values()) if isinstance(a, Config)]
+        except ValueError:
+            raise ValueError(
+                f"A config argument is required but was not found for {func.__module__ if hasattr(func, '__module__') else ''}: {func.__name__}"
+            )
+        if config.stash[stash_keys.VALIDATION_TEST_MODE] == Mode.DIAGNOSTIC:
+            try:
+                return await func(*args, **kwargs)
+            except AssertionError:
+                logging.warning(traceback.format_exc())
+        else:
+            return await func(*args, **kwargs)
+
+    return decorator.decorator(wrapper, func)
```

Comment:
Thanks, I like the reorganization. Done.

Regarding the name, to me "allow" means that the mode is permitted, but here we're enforcing it. WDYT about "enforce" instead of "respect"?